core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -2031,11 +2088,7 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
       s"${KafkaConfig.AdvertisedListenersProp} cannot use the nonroutable meta-address 0.0.0.0. "+
       s"Use a routable IP address.")
 
-    // Ensure controller listeners are not in the advertised listeners list
-    require(!controllerListeners.exists(advertisedListeners.contains),
-      s"${KafkaConfig.AdvertisedListenersProp} cannot contain any of ${KafkaConfig.ControllerListenerNamesProp}")
-
```

Comment:
This is now checked above -- this exact check is done when running the broker role, but the check for when running just the controller role is actually now more constrained: advertised listeners must be empty in that case.